web/modules/custom/dpl_opening_hours/src/Model/OpeningHoursRepository.php

```diff
@@ -59,14 +59,26 @@ public function load(int $id): ?OpeningHoursInstance {
   /**
    * Load a collection of opening hours.
    *
+   * @param int|int[] $branchId
+   *   The branch ID or an array of branch IDs.
+   * @param \DateTimeInterface|null $fromDate
+   *   The start date.
+   * @param \DateTimeInterface|null $toDate
+   *   The end date.
+   * @param int|null $repetitionId
+   *   The repetition ID.
+   * @param int|null $categoryId
+   *   The category ID.
+   *
    * @return OpeningHoursInstance[]
    *   Opening hours instances which match the provided criteria.
    */
-  public function loadMultiple(int $branchId = NULL, \DateTimeInterface $fromDate = NULL, \DateTimeInterface $toDate = NULL, int $repetitionId = NULL, int $categoryId = NULL): array {
+  public function loadMultiple(int|array $branchId = NULL, \DateTimeInterface $fromDate = NULL, \DateTimeInterface $toDate = NULL, int $repetitionId = NULL, int $categoryId = NULL): array {
     $query = $this->connection->select(self::INSTANCE_TABLE, self::INSTANCE_TABLE)
       ->fields(self::INSTANCE_TABLE);
     if ($branchId) {
-      $query->condition('branch_nid', $branchId);
+      $operator = is_array($branchId) ? 'IN' : '=';
+      $query->condition('branch_nid', $branchId, $operator);
```

Comment:
Consider how the signature of this function needs to be updated.

One approach as you have implemented is to accept multiple types. If you follow this approach I think it would be preferable to convert the single value to an array and then always pass this with the the `IN` operator as the condition.

Another approach would be to change the signature to accept `array $branchIds = NULL` and then update callers to pass single value arrays.